draft-ietf-capport-architecture.xml

```diff
@@ -401,18 +398,54 @@
         canary URLs to identify the state captivity [reference Mariko
-        Kobayashi's survey]. While doing so, ICMP messages SHOULD also be sent,
+        Kobayashi's survey]. While doing so, Captive Portal Signals SHOULD also be sent,
         to activate work-flows in supporting devices.
         [TODO: give some thought to precise recommendations for backwards compatibility.]
       </t>
      </section>
-     <section anchor="section_icmp" title="ICMP/ICMP6">
+     <section anchor="section_signal" title="Captive Portal Signal">
+       <t>
+        User Equipment may send traffic outside the captive network prior to the Enforcement
+        device granting it access. The Enforcement Device rightly blocks or resets these
+        requests. However, lacking a signal from the Enforcement Device, the User Equipment
+        can only guess at whether it is captive. Consequently, allowing the Enforcement Device
+        to signal to the User Equipment that there is a problem with its connectivity may
+        improve the user's experience.
+       </t>
        <t>
-         ICMP messages have been selected for indicating to a sender that
-         packets could not be delivered for reason of a network policy (in
-         particular, captive portal). ICMP is already used to indicate reasons
-         that packets could not be delivered (network unreachable, port
-         unreachable, packet too large, etc.).
+        An Enforcement Device may also want to inform the User Equipment of a pending expiry
+        of its access to the external network, so providing the Enforcement Device the ability
+        to preemptively signal may be desireable.
+       </t>
+       <t>
+        A specific Captive Portal Signalling Protocol is out of scope for this document.
+        However, in order to ensure that future protocols fit into the architecture,
+        requirements for a Captive Portal Signalling Protocl follow:
+          <list style="numbers">  
+            <t>The notification SHOULD NOT be easy to spoof.
+               If an attacker can send spoofed notifications to the User Equipment,
+               they can cause the User Equipment to unnecessarily access the API. Rather than
+               relying solely on rate limits to prevent problems, a good protocol will strive
+               to limit the feasibility of such attacks.
+            </t>
+            <t>It SHOULD be possible to send the notification before the captive portal closes.
+               This will help ensure seemless connectivity for the user, as the User Equipment
+               will not need to wait for a network failure to refresh its login. On receipt of
+               preemptive notication, the User Equipment can prompt the user to refresh.
+             </t>
+             <t>The signal SHOULD be a simple binary protocol indicating that the user equipment
+                is either captive or soon to be captive.
```

Comment:
This isn't really a requirement, it's desiderata, sure, but the point I think that you wanted to capture is that the protocol not include any information other than a prompt to contact the API.